src/ORM/Rule/ExistsIn.php

```diff
@@ -98,6 +109,11 @@ public function __invoke(EntityInterface $entity, array $options)
             return true;
         }
 
+        if ($this->_options['allowSqlNulls'] === true
+            && $this->_checkPartialSchemaNulls($entity, $source) === true
+        ) {
+            return true;
```

Comment:
For the matter at hand:
I did at first only fix the issue (see the commit log). Problem: people expect things to work like this now and it is not backwards compatible. Because of that I have added the option. For 4.x or 5.x we can switch the flag to true by default and/or introduce a flag to re-enable the old behavior.

Personal Note: In case you will understand what others are saying you should join. We had some people from Japan even giving some talks. Love to see you next time!